src/RestApi/StoreApi/Routes/AbstractRoute.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * Abstract route.
+ *
+ * @internal This API is used internally by Blocks--it is still in flux and may be subject to revisions.
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\RestApi\StoreApi\Routes;
+
+defined( 'ABSPATH' ) || exit;
+
+use Automattic\WooCommerce\Blocks\RestApi\StoreApi\Schemas\AbstractSchema;
+
+/**
+ * AbstractRoute class.
+ */
+abstract class AbstractRoute implements RouteInterface {
+	/**
+	 * Schema class instance.
+	 *
+	 * @var AbstractSchema
+	 */
+	protected $schema;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param AbstractSchema $schema Schema class for this route.
+	 */
+	public function __construct( AbstractSchema $schema ) {
```

Comment:
I realize there may be some routes that don't need a controller. You _could_ make it optional via the signature here: `public function __construct( AbstractSchema $schema, ControllerInterface $controller = null );` but I'd prefer to see us _require_ a controller (because it would help enforce keeping the logic separate a bit - controllers _could_ be tested independently of the route class for complicated logic).

Reply:
I think the only endpoints that use a controller are the Cart ones. I'm not really happy with the Cart Controller itself and as you point out, not all endpoints need them. We could also consider using a AbstractCartRoute to house some of the cart functionality, since the endpoints are quite coupled to that controller anyway.

I think let's leave that out of this PR and come back around, possibly during cooldown.